WIP: Doc/update

hachi8833 opened the pull request:
Just to notify it is under construction of updating parser

hachi8833 committed:
Refactor parser.go first

hachi8833 committed:
Update parser to obtain instance/class methods from FuncDecl

Update commit id

adlerhsieh commented:
Thanks. I might not have time to go through PRs recently. If everything is working locally and documents are generated correctly, then it should be fine 👍

hachi8833 commented:
@adlerhsieh Sorry, I have looked into `retrieveMethodsFromNode`, but it looks too complicated for me 😢 
It might take more than half a month for the fix. 
I'd be very glad if you'd fix the parser.

hachi8833 committed:
Update retrieveMethodsFromNode

Update commit id

Update for loops in retrieveMethodsFromNode()

Fix variable names

Extract FnName and FnLine

Parse comments, params, returns

Remove redundant for loop

Remove redundant for loop

Remove redundant for loop

adlerhsieh commented:
Yeap, I'll check it when I got time. Thanks for reporting!

hachi8833 commented:
Finally updated `retrieveMethodsFromNode()`. 
Perhaps this should be working, but look like some api doc in Goby have been a bit inconsistent and needs fix.
